dll/win32/shell32/wine/control.c

```diff
@@ -821,26 +821,29 @@ Control_ShowAppletInTaskbar(CPlApplet* applet, UINT index)
 {
     ITaskbarList* pTaskbar = NULL;
 
-    SetWindowTextW(applet->hWnd, applet->info[index].name);
-    SendMessageW(applet->hWnd, WM_SETICON, ICON_SMALL, (LPARAM)applet->info[index].icon);
-
-    /* Add button to the taskbar */
-    ShowWindow(applet->hWnd, SW_SHOWMINNOACTIVE);
-
-    /* Activate the corresponding button in the taskbar */
-    CoInitialize(NULL);
-    if (CoCreateInstance(&CLSID_TaskbarList,
-                         NULL, CLSCTX_INPROC_SERVER,
-                         &IID_ITaskbarList,
-                         (LPVOID*)&pTaskbar) == S_OK)
+    if (GetParent(applet->hWnd) == NULL) // Is desktop the parent window?
```

Comment:
Nit: In order to avoid the useless indentation change, you could do:
```suggestion
    /* Try to add a taskbar button only if the applet's parent window is the desktop */
    if (GetParent(applet->hWnd) != NULL)
        return;
```
(then follows the original unchanged code).

Reply:
Yes